src/Maestro/Client/src/MaestroApiCredential.cs

```diff
@@ -51,26 +54,74 @@ public override ValueTask<AccessToken> GetTokenAsync(TokenRequestContext _, Canc
+        internal static MaestroApiCredential CreateUserCredential(string barApiBaseUri, string cachePath = "")
         {
             string appId = EntraAppIds[barApiBaseUri.TrimEnd('/')];
+            var requestContext = new TokenRequestContext(new string[] { $"api://{appId}/{USER_SCOPE}" });
 
-            // This is a usual credential obtained against an entra app through a browser sign-in
-            var credential = new InteractiveBrowserCredential(new InteractiveBrowserCredentialOptions
+            // This is a usual configuration for a credential obtained against an entra app through a browser sign-in
+            var credentialOptions = new InteractiveBrowserCredentialOptions
             {
                 TenantId = TENANT_ID,
                 ClientId = appId,
                 RedirectUri = new Uri("http://localhost"),
                 TokenCachePersistenceOptions = new TokenCachePersistenceOptions()
                 {
                     Name = "darc"
-                }
-            });
+                },
+            };
+
+            InteractiveBrowserCredential credential;
+            if (!string.IsNullOrEmpty(cachePath))
+            {
+                string authRecordPath = Path.Combine(cachePath, $"{AUTH_RECORD_PREFIX}-{appId}");
+                credential = GetCredentialFromCache(credentialOptions, requestContext, authRecordPath);
+            }
+            else
+            {
+                // In certain cases like MSBuild tasks we don't have a cache to work with and
+                // always prompt the user for concent
+                credential = new InteractiveBrowserCredential(credentialOptions);
+            }
 
-            var requestContext = new TokenRequestContext(new string[] { $"api://{appId}/{USER_SCOPE}" });
             return new MaestroApiCredential(credential, requestContext);
         }
 
+        /// <summary>
+        /// Create interactive credential from an authentication record stored in local cache
+        /// Authentication record is a set of app and user-specific metadata used by the library to authenticate
+        /// </summary>
+        private static InteractiveBrowserCredential GetCredentialFromCache(
+            InteractiveBrowserCredentialOptions credentialOptions,
+            TokenRequestContext requestContext,
+            string authRecordPath)
+        {
+            InteractiveBrowserCredential credential;
+
+            if (File.Exists(authRecordPath))
+            {
+                // Fetch existing authentication record to not promt the user for concent
+                using var authRecordStream = new FileStream(authRecordPath, FileMode.Open, FileAccess.Read);
+                var authRecord = AuthenticationRecord.Deserialize(authRecordStream);
+
+                credentialOptions.AuthenticationRecord = authRecord;
+
+                credential = new InteractiveBrowserCredential(credentialOptions);
```

Comment:
Should we try/catch the deserialization and delete the file? Otherwise people who get their file messed up (or outdated) will never get out of that situation

Reply:
Great idea, added